plugins/org.eclipse.emfcloud.modelserver.glsp.integration/src/org/eclipse/emfcloud/modelserver/glsp/operations/handlers/EMSChangeBoundsOperationHandler.java

```diff
@@ -16,22 +16,30 @@
 import org.eclipse.emfcloud.modelserver.glsp.notation.Shape;
 import org.eclipse.emfcloud.modelserver.glsp.notation.integration.EMSNotationModelServerAccess;
 import org.eclipse.emfcloud.modelserver.glsp.notation.integration.EMSNotationModelState;
+import org.eclipse.glsp.server.model.GModelState;
 import org.eclipse.glsp.server.operations.ChangeBoundsOperation;
 import org.eclipse.glsp.server.types.ElementAndBounds;
 import org.eclipse.glsp.server.types.GLSPServerException;
 
+import com.google.inject.Inject;
+
 public class EMSChangeBoundsOperationHandler
-   extends EMSBasicOperationHandler<ChangeBoundsOperation, EMSNotationModelState, EMSNotationModelServerAccess> {
+   extends EMSBasicOperationHandler<ChangeBoundsOperation, EMSNotationModelServerAccess> {
+
+   @Inject
+   protected GModelState modelState;
```

Comment:
Do we need to inject the state again in the actual handler or is the state in the basic class, in this case the EMSBasicOperationHandler enough?
If it is enough, would it make sense to add a method to get the EMSNotationModelState directly from the EMSBasicOperationHandler? 
Same question applies to the EMSBasicActionHandler

Reply:
Thanks for pointing that out, that's a good point! I addressed your comment in a separate commit.